internal/tofu/node_provider.go

```diff
@@ -56,6 +57,12 @@ func (n *NodeApplyableProvider) ValidateProvider(ctx EvalContext, provider provi
 
 	configBody := buildProviderConfig(ctx, n.Addr, n.ProviderConfig())
 
+	// If a provider configuration is of type TestProviderConfig, then skip the validation,
+	// and its value will be evaluated before the Plan operation.
+	if _, ok := configBody.(*configs.TestProviderConfig); ok {
```

Comment:
can you please add a unit test for this case please? This method is pretty well tested and it helps us avoid regressions in the future. Thanks!